tests/render/test_render.py

```diff
@@ -0,0 +1,86 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from phabricatoremails.render.events.phabricator import RevisionUpdated
+from phabricatoremails.render.render import Render
+from tests.mock_thread_store import MockThreadStore
+from tests.render.mock_template import MockTemplateStore
+
+BODY = {
+    "isReadyToLand": True,
+    "newChangesLink": "link",
+    "affectedFiles": [{"path": "/file", "change": "added"}],
+    "reviewers": [
+        {
+            "name": "reviewer",
+            "isActionable": True,
+            "status": "requested-changes",
+            "recipients": [
+                {"timezoneOffset": -25200, "email": "reviewer@mail", "isActor": False}
+            ],
+        }
+    ],
+}
+
+
+def _create_public_event(revision_id: int = 1):
+    return {
+        "isSecure": False,
+        "eventKind": RevisionUpdated.KIND,
+        "timestamp": 123,
+        "actorName": "actor",
+        "revision": {"revisionId": revision_id, "name": "revision", "link": "link"},
+        "body": BODY,
+    }
```

Comment:
This and the below should be fixtures:
```suggestion
@pytest.fixture
def create_public_event():
    def _create_public_event(revision_id=1):
        return {
            "isSecure": False,
            "eventKind": RevisionUpdated.KIND,
            "timestamp": 123,
            "actorName": "actor",
            "revision": {"revisionId": revision_id, "name": "revision", "link": "link"},
            "body": BODY,
        }
    return _create_public_event
```

Reply:
How does piping functions through the `pytest.fixture` mechanism, rather than directly calling the function, improve the tests?

Reply:
> Fixtures are automatically imported

Implicit imports are not necessarily better than explicit imports.

> are initialized before every test

Helper functions shouldn't use global state, and those that I've set up in the tests here don't If you call `_create_public_event(...)`, you get a _freshly initialized_ public event

> can be parameterized

Functions can be parameterized.

> You can avoid some common gotchas by using fixtures, as well as keep your code clean.

I think that fixtures that are coupled to test lifecycle are useful (e.g.: creating temporary files/directories), but I don't think that all helper functions should be refactored into fixtures.